mdonatello/mdonatello.py

```diff
@@ -9,3 +9,138 @@
+            )
+            d.FinishDrawing()
+            svg = d.GetDrawingText()
+            return HTML(svg)
+        
+        if show_atom_indices:
+            for atom in ag.GetAtoms():
+                atom.SetProp("atomNote", str(atom.GetIdx()))
+        else:
+            for atom in ag.GetAtoms():
+                atom.ClearProp("atomNote")
+        
+        bio = BytesIO()
+        img = Draw.MolToImage(ag)
+        img.save(bio, format="png")
+        html = f'<img src="data:image/png;base64,{base64.b64encode(bio.getvalue()).decode()}" />'
+        return HTML(html)
+        
+    def update_display(self, _=None):
+        self.hbond_props_checkbox.description = self.get_hbond_description()
+        smiles = Chem.MolToSmiles(self.ag_noh)
+        
+        children = [
+            self.display_molecule(self.ag_noh, self.show_atom_indices_checkbox.value, self.highlight_aromatic_checkbox.value),
+            HTML(f"<h3 style='margin: 0;'>SMILES: {smiles}</h3>")
+        ]
+        
+        if self.physiochem_props_checkbox.value:
+            children.extend([
+                self.display_molecular_weight(self.ag_noh),
+                self.display_logp(self.ag_noh),
+                self.display_tpsa(self.ag_noh),
+                self.display_rotatable_bonds(self.ag_noh)
+            ])
+            
+        if self.hbond_props_checkbox.value:
+            children.extend([
+                self.display_num_h_donors(self.ag_noh),
+                self.display_num_h_acceptors(self.ag_noh)
+            ])
+        
+        self.output_molecule.children = children
+        
+    def display_molecular_weight(self, ag):
+        mw = Descriptors.MolWt(ag)
+        return HTML("<p style='margin: 0; margin-left: 100px;'>Molecular Weight: {:.2f} g/mol</p>".format(mw))
+        
+    def display_logp(self, ag):
+        logp = Descriptors.MolLogP(ag)
+        return HTML("<p style='margin: 0; margin-left: 100px;'>LogP: {:.2f}</p>".format(logp))
+
+    def display_num_h_donors(self, ag):
+        num_h_donors = Descriptors.NumHDonors(ag)
+        return HTML("<p style='margin: 0; margin-left: 100px;'>Number of H-Bond Donors: {:.0f}</p>".format(num_h_donors))
+
+    def display_num_h_acceptors(self, ag):
+        num_h_acceptors = Descriptors.NumHAcceptors(ag)
+        return HTML("<p style='margin: 0; margin-left: 100px;'>Number of H-Bond Acceptors: {:.0f}</p>".format(num_h_acceptors))
+    
+    def display_tpsa(self, ag):
+        tpsa = Descriptors.TPSA(ag)
+        return HTML("<p style='margin: 0; margin-left: 100px;'>Topological Polar Surface Area (TPSA): {:.2f} Å²</p>".format(tpsa))
+        
+    def display_rotatable_bonds(self, ag):
+        rotatable_bonds = Descriptors.NumRotatableBonds(ag)
+        return HTML("<p style='margin: 0; margin-left: 100px;'>Number of Rotatable Bonds: {:.0f}</p>".format(rotatable_bonds))
+        
+    def save_selected_molecule(self, _):
```

Comment:
probably filename as an argument here right?  In general if you want to use print, you probably want to instead do a logging call